lib/filesystem/CFilesystemLoader.cpp

```diff
@@ -173,19 +176,26 @@ std::unordered_map<ResourcePath, boost::filesystem::path> CFilesystemLoader::lis
 				filename = it->path().filename();
 
 			std::string resName;
+
+#ifdef VCMI_WINDOWS
+			std::string filenameUtf8 = boost::locale::conv::utf_to_utf<char>(filename.native());
+#else
+			std::string filenameUtf8 = filename.string();
+#endif
+
```

Comment:
Already copy-pasted once, and it's possible that there are other, not yet found cases.
Perhaps move to TextOperations file?
```cpp
std::string TextOperations::filesystemPathToUtf8(const boost::filesystem::path & path)
{
#ifdef VCMI_WINDOWS
	return boost::locale::conv::utf_to_utf<char>(filename.native());
#else
	return filename.string();
#endif
}
```

And possibly do the same for the opposite conversion - from utf8 string to filesystem path